src/app/wallet/address-book/address-book.component.ts

```diff
@@ -1,15 +1,97 @@
 import { Component, OnInit } from '@angular/core';
 
+import { RPCService } from '../../core/rpc/rpc.service';
+
 @Component({
   selector: 'app-address-book',
   templateUrl: './address-book.component.html',
   styleUrls: ['./address-book.component.scss']
 })
 export class AddressBookComponent implements OnInit {
 
-  constructor() { }
+  label: string = '';
+  address: string;
+  validAddress: boolean = undefined;
+
+  openNewAddressModal: boolean = false;
+
+  constructor(private _rpc: RPCService) { }
 
   ngOnInit() {
+    document.onkeydown = evt => {
+      if (evt.key.toLowerCase() === 'escape') {
+        this.closeNewAddress();
+      }
+    }
+  }
+
+  openNewAddress() {
+    this.openNewAddressModal = true;
+  }
+
+  closeNewAddress() {
+    this.openNewAddressModal = false;
+  }
+
+  addAddressToBook() {
+    if (this.validAddress && this.label !== undefined) {
+      this._rpc.call(this, 'manageaddressbook', ['add', this.address, this.label], this.rpc_callbackAddAddressToBook);
+      this.address = undefined;
+      this.validAddress = undefined;
+      this.label = '';
+      this.closeNewAddress();
+    } else {
+      alert('Please enter a valid address!');
+    }
+  }
+
+  rpc_callbackAddAddressToBook(json: Object) {
+    if (json['result'] === 'success') {
+      alert('Address successfully added to the addressbook!');
+
+      // TODO: remove specialPoll!
+      this._rpc.specialPoll();
+    }
+  }
+
+  verifyAddress() {
+    if (this.address === undefined || this.address === '') {
+      this.validAddress = undefined;
+      return;
+    }
+
+    const ret = false;
+    if ((this.address.indexOf('p') === 0) === false) { // does not start with p
```

Comment:
This validation won't work for mainnet.

we should do a base58 check oninput of the address as well:
```

var base58 = {
    base58Chars :"123456789ABCDEFGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz",

    check: function(field)
    {
        var el = $(field);
        var value = el.val();

        for (var i = 0, len = value.length; i < len; ++i)
            if (base58.base58Chars.indexOf(value[i]) == -1) {
                el.css("background", "#E51C39").css("color", "white");
                return false;
            }

        el.css("background", "").css("color", "");
        return true;
    }
}
```

That's what happend in shadow on input.
We could also do a regular expression to validate the address with 'pP' and 'PT'